opencensus/trace/tracer.py

```diff
@@ -73,7 +73,7 @@ def should_sample(self):
         :rtype: bool
         :returns: Whether to trace the request or not.
         """
-        return self.span_context.trace_options.enabled \
+        return self.span_context.trace_options.get_enabled() \
             or self.sampler.should_sample(self.span_context.trace_id)
```

Comment:
Related to this PR, `should_sample` needs to take the trace options as well. For example, the way the probability sampler should work is that if trace options is true then return true, else check against the configured probability. This way if a remote parent is sampled the children will be too, but still allowing for a sampler to override that.